go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -508,9 +516,6 @@ func (app *rootHashApplication) processRoundTimeout(ctx *tmapi.Context, state *r
 		return fmt.Errorf("failed to finalize block: %w", err)
 	}
 
-	if err = state.SetRuntimeState(ctx, rtState); err != nil {
```

Comment:
:information_source: this is moved into the above tryFinalizeBlock. all two callsites of tryFinalizeBlock would do this

---

but this breaks some symmetry, where we obtain rtState at this level, but we do the 'cleanup' procedures one level down inside tryFinalizeBlock